### PR TITLE
Unify data preparation

### LINKED ISSUE #7: Unify data preprocessing steps
Steps:
- Ensure that the input is transformed to DataFrame, set the dtype to category, such that LGBM can handle them
- Make a generic preprocess function for X argument. Ensure it is always transformed to df
- Test each feature that should allow for categorical variables, whether it accepts it e.g. SHAPResemblanceModel, MetricVolatility etc.

### PR DESCRIPTION
**Pull request by Matgrb**
Closes #7 
- Unifies Data preprocessing by adding preprocess data and preprocess labels functions
- Adds unified Verbose and column_names paramer to classes that use datasets to train models
- Adds docstrings on new parameters
- Adds tests with Lightgbm on messy data for each class that should handle categorical variable
- Disables LightGBM tests in Github Actions, due to difficulty in installing properly LightGBM on Mac with Github actions.

### CODE REVIEW

File: .github/workflows/publish_to_pypi.yml [hunk 1]
```diff
@@ -21,6 +21,7 @@ jobs:
     - name: Make sure unit tests succeed
       run: |
         pip install .
+        pip install -r requirements_test.txt
```
Comment (timvink) [inline .github/workflows/publish_to_pypi.yml:24]: I recommend using tests/test_requirements.txt

File: probatus/interpret/shap_dependence.py [hunk 2]
```diff
@@ -48,16 +48,31 @@ class TreeDependencePlotter:
...
 
+            verbose (int, optional):
+                Controls verbosity of the output:
+
+                - 0 - nether prints nor warnings are shown
+                - 1 - 50 - only most important warnings regarding data properties are shown (excluding SHAP warnings)
+                - 51 - 100 - shows most important warnings, prints of the feature removal process
+                - above 100 - presents all prints and all warnings (including SHAP warnings).
+        """
+        self.model = model
```
Comment (timvink) [inline probatus/interpret/shap_dependence.py:67]: `model` is not in docstring

File: probatus/metric_volatility/volatility.py [hunk 3]
```diff
@@ -489,7 +519,7 @@ class BootstrappedVolatility(TrainTestVolatility):
     from sklearn.ensemble import RandomForestClassifier
     from probatus.metric_volatility import BootstrappedVolatility
     X, y = make_classification(n_features=4)
-    RandomForestClassifier()
+    clf = RandomForestClassifier()
```
Comment (timvink) [inline probatus/metric_volatility/volatility.py:522]: Good catch. I can show you how to write unit tests to make sure the code snippets in the docstring actually work. Feel free to plan 30min call

File: probatus/sample_similarity/resemblance_model.py [hunk 4]
```diff
@@ -99,38 +107,31 @@ def fit(self, X1, X2, column_names=None):
...
+        self.y = preprocess_labels(self.y, y_name='y', index=self.X.index, verbose=self.verbose)
+
         # Reinitialize variables in case of multiple times being fit
         self._init_output_variables()
 
-        self.X_train, self.X_test, self.y_train, self.y_test = train_test_split(self.X, self.y,test_size=self.test_prc,
+        self.X_train, self.X_test, self.y_train, self.y_test = train_test_split(self.X, self.y, test_size=self.test_prc,
                                                                                 random_state=self.random_state,
-                                                                                stratify=self.y)
+                                                                                shuffle=True, stratify=self.y)
```
Comment (timvink) [inline probatus/sample_similarity/resemblance_model.py:134]: good catch. small change, good to make a list of all these small tweaks in the changelog ;)

File: probatus/sample_similarity/resemblance_model.py [hunk 5]
```diff
@@ -414,28 +423,32 @@ class SHAPImportanceResemblance(BaseResemblanceModel):
     <img src="../img/sample_similarity_shap_summary.png" width="320" />
     """
 
-    def __init__(self, model, iterations=100, test_prc=0.25, n_jobs=1, random_state=42):
+    def __init__(self, model, test_prc=0.25, n_jobs=1, verbose=0, random_state=42):
```
Comment (timvink) [inline probatus/sample_similarity/resemblance_model.py:426]: not familiar enough with code base, but are you sure you can remove this parameter?

**Review by timvink — COMMENTED** [review]
Lots good!

File: probatus/sample_similarity/resemblance_model.py (hunk 5 above)
Comment (Matgrb) [reply probatus/sample_similarity/resemblance_model.py:426]: Yes, this parameter is not used in any way in ShapResemblance, it is only used in permutation one.
